From 72d9dd355e1c7b30a095602ba942261282664269 Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Fri, 29 Dec 2017 14:34:06 +0800
Subject: [PATCH] ARM: dts: rockchip: update rk808's regulators for rk3036
 kylin

According to the actual schematic designed by kylin board, update and
rename the regulators for rk808 node information.

Especially gpu regulator voltage, the schematic didn't have this
regulator, this regulaor should be applied by cpu regulator since the
cpu/gpu/ddr are belong to the same logic power supply.

Change-Id: I39e4cf18969391da396cc775f8660701e42977bd
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 arch/arm/boot/dts/rk3036-kylin.dts | 90 +++++++++++++++---------------
 1 file changed, 44 insertions(+), 46 deletions(-)

diff --git a/arch/arm/boot/dts/rk3036-kylin.dts b/arch/arm/boot/dts/rk3036-kylin.dts
index ec93e8b4a524..3bea83fe7fc2 100644
--- a/arch/arm/boot/dts/rk3036-kylin.dts
+++ b/arch/arm/boot/dts/rk3036-kylin.dts
@@ -182,7 +182,20 @@
 
 &gpu {
 	status = "okay";
-	mali-supply = <&vdd_gpu>;
+	mali-supply = <&vdd_cpu>;
+};
+
+/*
+ * Just as GPU's power supply is provided by cpu regulator, and fixed-frequency
+ * on CPU.
+ */
+&gpu_opp_table {
+	opp-200000000 {
+		opp-microvolt = <1150000>;
+	};
+	opp-400000000 {
+		opp-microvolt = <1150000>;
+	};
 };
 
 &hdmi {
@@ -211,40 +224,25 @@
 		clock-output-names = "xin32k", "rk808-clkout2";
 
 		vcc1-supply = <&vcc_sys>;
-		vcc2-supply = <&vcc_sys>;
 		vcc3-supply = <&vcc_sys>;
 		vcc4-supply = <&vcc_sys>;
 		vcc6-supply = <&vcc_sys>;
 		vcc7-supply = <&vcc_sys>;
-		vcc8-supply = <&vcc_18>;
-		vcc9-supply = <&vcc_io>;
+		vcc9-supply = <&vcc_sys>;
 		vcc10-supply = <&vcc_io>;
-		vcc11-supply = <&vcc_sys>;
-		vcc12-supply = <&vcc_io>;
-		vddio-supply = <&vccio_pmu>;
+		vcc11-supply = <&vcc_io>;
+		vddio-supply = <&vcc_io>;
 
 		regulators {
 			vdd_cpu: DCDC_REG1 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <1000000>;
-				regulator-max-microvolt = <1350000>;
+				regulator-min-microvolt = <1150000>;
+				regulator-max-microvolt = <1150000>;
 				regulator-name = "vdd_arm";
 				regulator-state-mem {
 					regulator-on-in-suspend;
-					regulator-suspend-microvolt = <1050000>;
-				};
-			};
-
-			vdd_gpu: DCDC_REG2 {
-				regulator-always-on;
-				regulator-boot-on;
-				regulator-min-microvolt = <850000>;
-				regulator-max-microvolt = <1250000>;
-				regulator-name = "vdd_gpu";
-				regulator-state-mem {
-					regulator-on-in-suspend;
-					regulator-suspend-microvolt = <1000000>;
+					regulator-suspend-microvolt = <1150000>;
 				};
 			};
 
@@ -269,62 +267,62 @@
 				};
 			};
 
-			vccio_pmu: LDO_REG1 {
+			vcca_33: LDO_REG1 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
-				regulator-name = "vccio_pmu";
+				regulator-name = "vcca_33";
 				regulator-state-mem {
 					regulator-on-in-suspend;
 					regulator-suspend-microvolt = <3300000>;
 				};
 			};
 
-			vcc_tp: LDO_REG2 {
+			vout1: LDO_REG2 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
-				regulator-name = "vcc_tp";
+				regulator-name = "vout1";
 				regulator-state-mem {
 					regulator-off-in-suspend;
 				};
 			};
 
-			vdd_10: LDO_REG3 {
+			vdd_11: LDO_REG3 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1000000>;
 				regulator-max-microvolt = <1000000>;
-				regulator-name = "vdd_10";
+				regulator-name = "vdd_11";
 				regulator-state-mem {
 					regulator-on-in-suspend;
 					regulator-suspend-microvolt = <1000000>;
 				};
 			};
 
-			vcc18_lcd: LDO_REG4 {
+			vout3: LDO_REG4 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <1800000>;
-				regulator-name = "vcc18_lcd";
+				regulator-max-microvolt = <3400000>;
+				regulator-name = "vout3";
 				regulator-state-mem {
 					regulator-on-in-suspend;
 					regulator-suspend-microvolt = <1800000>;
 				};
 			};
 
-			vccio_sd: LDO_REG5 {
+			vout4: LDO_REG5 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-name = "vccio_sd";
+				regulator-max-microvolt = <3400000>;
+				regulator-name = "vout4";
 				regulator-state-mem {
 					regulator-on-in-suspend;
-					regulator-suspend-microvolt = <3300000>;
+					regulator-suspend-microvolt = <1800000>;
 				};
 			};
 
@@ -332,7 +330,7 @@
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <2500000>;
+				regulator-max-microvolt = <3400000>;
 				regulator-name = "vout5";
 				regulator-state-mem {
 					regulator-on-in-suspend;
@@ -340,43 +338,43 @@
 				};
 			};
 
-			vcc_18: LDO_REG7 {
+			vout2: LDO_REG7 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <1800000>;
-				regulator-name = "vcc_18";
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <2500000>;
+				regulator-name = "vout2";
 				regulator-state-mem {
 					regulator-on-in-suspend;
 					regulator-suspend-microvolt = <1800000>;
 				};
 			};
 
-			vcca_codec: LDO_REG8 {
+			vcc18: LDO_REG8 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
-				regulator-name = "vcca_codec";
+				regulator-name = "vcc18";
 				regulator-state-mem {
 					regulator-on-in-suspend;
 					regulator-suspend-microvolt = <1800000>;
 				};
 			};
 
-			vcc_wl: SWITCH_REG1 {
+			vswout1: SWITCH_REG1 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-name = "vcc_wl";
+				regulator-name = "vswout1";
 				regulator-state-mem {
 					regulator-on-in-suspend;
 				};
 			};
 
-			vcc_lcd: SWITCH_REG2 {
+			vswout2: SWITCH_REG2 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-name = "vcc_lcd";
+				regulator-name = "vswout2";
 				regulator-state-mem {
 					regulator-on-in-suspend;
 				};
-- 
2.35.3

